bugbot/rules/inactive_revision.py

```diff
@@ -0,0 +1,328 @@
+                        "new": "true",
+                    }
+                ],
+            }
+
+    def _find_last_action(self, revision_id):
+        details = self._fetch_revisions([revision_id])
+
+        if not details:
+            return None, None
+
+        revision = details[0]
+        author_phid = revision["fields"]["authorPHID"]
+        reviewers = [
+            reviewer["reviewerPHID"]
+            for reviewer in revision["attachments"]["reviewers"]["reviewers"]
+        ]
+
+        transactions = self._fetch_revision_transactions(revision["phid"])
+        last_transaction = transactions[0] if transactions else None
+
+        if last_transaction:
+            last_action_by_phid = last_transaction["authorPHID"]
+            if last_action_by_phid == author_phid:
+                last_action_by = "author"
+            elif last_action_by_phid in reviewers:
+                last_action_by = "reviewer"
+            else:
+                last_action_by = "other"
+        else:
+            last_action_by = "unknown"
+
+        return last_action_by, last_transaction
+
+    def _get_revisions_with_inactive_action(self, rev_ids: list) -> Dict[int, dict]:
+        revisions: List[dict] = []
+
+        for _rev_ids in Connection.chunks(rev_ids, PHAB_CHUNK_SIZE):
+            for revision in self._fetch_revisions(_rev_ids):
+                if (
+                    len(revision["attachments"]["reviewers"]["reviewers"]) == 0
+                    or revision["fields"]["status"]["value"] != "needs-review"
+                    or revision["fields"]["isDraft"]
+                ):
+                    continue
+
+                _, last_transaction = self._find_last_action(revision["id"])
+
+                if (
+                    last_transaction
+                    and last_transaction["dateCreated"] < self.patch_activity_limit
+                ):
+                    reviewers = [
+                        {
+                            "phid": reviewer["reviewerPHID"],
+                            "is_group": reviewer["reviewerPHID"].startswith(
+                                "PHID-PROJ"
+                            ),
+                            "is_blocking": reviewer["isBlocking"],
+                            "is_accepted": reviewer["status"] == "accepted",
+                            "is_resigned": reviewer["status"] == "resigned",
+                        }
+                        for reviewer in revision["attachments"]["reviewers"][
+                            "reviewers"
+                        ]
+                    ]
+
+                    if any(
+                        reviewer["is_group"] or reviewer["is_accepted"]
+                        for reviewer in reviewers
+                    ) and not any(
+                        not reviewer["is_accepted"]
+                        for reviewer in reviewers
+                        if reviewer["is_blocking"]
+                    ):
```

Comment:
This condition is a bit confusing with the double negation, could you use `all` instead to avoid the double negation?

Reply:
Good point. Done here: f739f18e1e2300ed69dca06e6efc217e8dce93e4